src/hypothesis/strategies.py

```diff
@@ -885,10 +875,7 @@ def decimals(min_value=None, max_value=None,
         # Fixed-point decimals are basically integers with a scale factor
 
         def try_quantize(d):
-            try:
-                return d.quantize(factor)
-            except InvalidOperation:  # pragma: no cover
-                return None
+            return d.quantize(factor)
```

Comment:
Without the try/except, this could be an inline lambda - but fixing decimals should be a separate pull and include tests showing that it works and why.